src/Common/ZooKeeper/ZooKeeper.h

```diff
@@ -2,6 +2,7 @@
 
 #include "Types.h"
 #include <Poco/Util/LayeredConfiguration.h>
+#include <optional>
```

Comment:
Note: I want to add a query to expose whether connection is within local az or not. There're two ways to do this:

- Add another field `system.zookeeper_connection` such as `connected_to_local_az`.
- The monitoring system can write a query as below. 

```sql
 :) select 'server-az-eg-us-west-2b' AS f  where f in (select value from system.zookeeper where path = '/keeper' and name ='api_version');

SELECT 'server-us-west-2b' AS f                     
WHERE f IN (
    SELECT value                                                                                                                                                                                                                            
    FROM system.zookeeper
    WHERE (path = '/keeper') AND (name = 'availability_zone')
)           
                                                           
Query id: f4aedc26-259d-49e6-b85f-659a282693ac

```

I am slightly prefer the first approach (add colum for `system.zookeeper_connection`) because that avoids additional request to keeper. WDTY? @antonio2368 @CheSema

Reply:
@kgoralski mentioned that we can also use `system.metrics` similar to what he did in https://github.com/ClickHouse/ClickHouse/pull/52378/files

Reply:
I would prefer to find fields `server_availability_zone` and `client_availability_zone` in table `system. zookeeper_connection`.

Reply:
I also agree with `system. zookeeper_connection`

Reply:
Done. I ended up doing both. Was told it'll be easier to setup using metrics in our case.